File: core/src/adictive/games/utils/MyButton.java
package adictive.games.utils;

import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.badlogic.gdx.graphics.g2d.TextureRegion;
import com.badlogic.gdx.math.Rectangle;

/**
 * Created by arielalvarez on 2/4/18.
 */

public class MyButton extends MyView {
    private OnButtonClick action;
    private boolean isTouching;

    public MyButton(String imgName, int side, int x, int y, OnButtonClick action) {
        super(new TextureRegion(new Texture(Gdx.files.internal("data/" + imgName + ".png"))));
        float ratio = (float) this.texture.getRegionHeight() / (float) this.texture.getRegionWidth();
        this.pos = new Rectangle(x, y, side, side * ratio);
        this.action = action;
    }

    public boolean isTouching() {
        return isTouching;
    }

    public void render(SpriteBatch batch) {
        boolean lastPositionTouchesButton = pos.contains(
                Gdx.input.getX() + pos.width / 2,
                Gdx.graphics.getHeight() - Gdx.input.getY() + pos.height / 2
        );

        boolean shouldAct = lastPositionTouchesButton && isTouching && !Gdx.input.isTouched() && action != null;

        isTouching = (lastPositionTouchesButton && Gdx.input.justTouched()) || (isTouching && Gdx.input.isTouched());

        ViewUtils.draw(batch, isTouching, pos, texture);

        if (shouldAct) {
            action.onClick();
        }
    }

}

File: core/src/adictive/games/utils/MyButton.kt
package adictive.games.utils

import com.badlogic.gdx.Gdx
import com.badlogic.gdx.graphics.Texture
import com.badlogic.gdx.graphics.g2d.SpriteBatch
import com.badlogic.gdx.graphics.g2d.TextureRegion
import com.badlogic.gdx.math.Rectangle

/**
 * Created by arielalvarez on 2/4/18.
 */

class MyButton(imgName: String, side: Int, x: Int, y: Int, private val action: () -> Unit) : MyView(TextureRegion(Texture(Gdx.files.internal("data/$imgName.png")))) {
    var isTouching: Boolean = false
        private set

    init {
        val ratio = this.texture.regionHeight.toFloat() / this.texture.regionWidth.toFloat()
        this.pos = Rectangle(x.toFloat(), y.toFloat(), side.toFloat(), side * ratio)
    }

    override fun render(batch: SpriteBatch) {
        val lastPositionTouchesButton = pos!!.contains(
                Gdx.input.x + pos!!.width / 2,
                Gdx.graphics.height - Gdx.input.y + pos!!.height / 2
        )

        val shouldAct = lastPositionTouchesButton && isTouching && !Gdx.input.isTouched

        isTouching = lastPositionTouchesButton && Gdx.input.justTouched() || isTouching && Gdx.input.isTouched

        ViewUtils.draw(batch, isTouching, pos!!, texture)

        if (shouldAct) {
            action.invoke()
        }
    }

}